File: programming-fundamentals-i/Chapter 6/Circle.java
public class Circle
	 {
		 private double radius;
		 private final double PI = 3.14159;
		 //constructor with radius as argument
		 public Circle(double r)
		 {
			 radius = r;
		 }
		 //constructor with no arguments
		 public Circle()
		 {
			 radius = 0.0;
		 }
		 public void setRadius(double r)
		 {
			 radius = r;
		 }
		 public double getRadius()
		 {
			 return radius;
		 }
		 public double getArea()
		 {
			 return PI * radius * radius;
		 }
		 public double getDiameter()
		 {
			 return radius * 2;
		 }
		 public double getCircumference()
		 {
			 return 2 * PI * radius;
		 }
	 }

File: Programming-Fundamentals-II/Project-5/Circle.cpp
// Implementation file for the Circle class.
#include "Circle.h"
#include <cstdlib>
#include <iostream>


Circle::Circle() //default constructor
{
	radius = 0.0;
}

Circle::Circle(double length) //constructor that takes radius as input
{
	if (length >= 0)
	{
		radius = length;
	}
	else
	{
		std::cout << "Invalid radius.\n";
		exit(EXIT_FAILURE);
	}
}

void Circle::setRadius(double length)
{
	if (length >= 0)
	{
		radius = length;
	}
	else
	{
		std::cout << "Invalid radius.\n";
		exit(EXIT_FAILURE);
	}
}
double Circle::getRadius()
{
	return radius;
}
double Circle::getArea()
{
	return pi * radius * radius;
}
double Circle::getDiameter()
{
	return 2 * radius;
}
double Circle::getCircumference()
{
	return 2 * pi * radius;
}
